### DISCUSSION TITLE
[Feature]: Webdav support (import/export)

### DISCUSSION THREAD

**Discussion by matthinc** [immich-app/immich#1687] [Feature Request]
### Feature detail

It would be nice to have support for import/export of assets from/to platforms supporting webdav (for example Nextcloud).

I don't see this as a high priority feature, just something to consider in the future.

### Platform

Server

**Comment by stokito** [comment]
The WebDAV would be nice because users may install Lighttpd + DAV module on their OpenWrt router and have a "cloud" for free.

**Comment by Majestic7979** [comment]
I see this absolutely as a high priority thing because exposing the library as a folder to any OS that supports WebDAV or any service (like Nextcloud) avoids duplication of potentially terabytes of data. It also cuts the need for a dedicated per-OS desktop sync client. This should 1000000% be treated as a priority. One of the reasons I am moving away from GP is being tired of having to maintain two sets of data which aren't for backup purposes. I also hate that GP lowers video quality when doing a TakeOut operation. Immich with WebDAV would be a killer combo.

**Reply by AngelaDMerkel** [reply]
@Majestic7979 Off topic, but does takeout alter the original video quality on export?

**Reply by Majestic7979** [reply]
Yes, it compresses the file until it basically becomes unusable junk. Google is evil.

**Comment by Happyfeet01** [comment]
Hello,

The discussion is not yet so lively. I would like to transfer my pictures either via the Windows computer or from my Nextcloud. But I see my options there.

The import from the storage path doesn't seem to work at the moment.

**Comment by sanderdam** [comment]
Hi,

For me this also is a missing feature of Immich.

My family who most are Windows users are used to organize their photos in folders just in Windows Explorer.
When Immich supports webdav, I can create a Network Connection to Immich so that they can organize photos using Explorer.
But, then the photos are still in Immich and so we can use all those features for sharing.

**Reply by AngelaDMerkel** [reply]
This would also have the positive side effect of enabling an application like Lightroom to publish directly to these directories and maintain the images in them. This would reduce the risk of out of date versions or duplicates which are not hash-identical

**Reply by mayeths** [reply]
Agreed. Supporting WebDAV is an elegant way to share photos inside the family, where they can view photos as normal files.

**Comment by Happyfeet01** [comment]
Maybe an Windows Client would helpful in the feature

**Reply by Majestic7979** [reply]
No. Please do not start diluting a clear request with OS specific stuff. WebDAV is platform-agnostic. IT JUST WORKS. This isn't about a desktop app, it's about WebDAV, it's clear in the title.

**Reply by aisbergde** [reply]
I agree. Web works fine on Windows. Web works even on mobile, some features of the web client are even better, than the mobile client. I would even prefer a well working web version on mobile over a feature limited mobile app.

**Comment by nirmal-mewada** [comment]
This functionality appears to offer significant value, especially when syncing local albums to immich using tools such as rclone.

**Reply by AngelaDMerkel** [reply]
This wouldn't really be necessary given rsync can offer the same functionality between two systems. I currently use rsync to do exactly this

**Reply by Majestic7979** [reply]
Are you syncing all your cleavage selfies?
![fashionbeauty__suze-images-suze080603-1](https://github.com/immich-app/immich/assets/128995501/a24d1f4e-a753-488a-8ffb-f1759fb1ebae)

**Reply by nirmal-mewada** [reply]
That does with file system, you still need to schedule library scans and also can not manage albums with your folder structure. see https://rclone.org/googlephotos/
This would be more robust then doing that with rsync "outside"  on file system.

**Comment by EuleMitKeule** [comment]
I also would love this feature. In an ideal world I could use Nextcloud to sync my media files between my devices to store them at a central location and Immich would recognize them and allow me to view, edit and delete the files stored in Nextcloud.

**Comment by johnmmcgee** [comment]
I too would like this.  Thank you.

**Comment by brianonymous** [comment]
Here to add support. I'm running photoprism in parallel for the webdav endpoint. It is probably the only aspect remaining before I delete the   photoprism container. I could probably implement a standalone webdav endpoint, but I only use it for photo sync, so it makes sense to me to keep it in my primary photo manager.

**Reply by outpoints** [reply]
That's exactly my stack as well. If the Immich app would upload my library correctly in the background this wouldn't be as much of an issue :p

**Comment by johnmmcgee** [comment]
I also would love to see this feature implemented.

**Comment by johnnyq** [comment]
Yeah same here I would love to have WebDAV Support for my parents and myself

**Comment by ruskrava** [comment]
Присоединяюсь. Это было бы очень полезно для интеграции например с Nextcloud. Думаю это важно.

**Comment by barrybecker4** [comment]
I would also like to see WebDAV support in Immich. I just set it up with PhotoPrism and it works great.
I haven't tried it yet, but maybe a workaround is to use the WebDAV integration in NextCloud, then point Immich to the directory in NextCloud. I am using Umbrel Home.

**Comment by kthornbloom** [comment]
Definitely interested as well! I spent considerable time setting up Immich once, but it seemed to be geared towards new galleries. Adding a pre-existing one was painful (at least at the time), and even then it had to import everything. Just reading from webdav would be great.

**Comment by kasparlov** [comment]
I would also love to have webdav support, similar to photoprism. I'm using a hosted nextcloud and it would be so nice to just link immich to the photo folder.

**Comment by omayhemo** [comment]
I came looking for WebDav support, so I can integrate with NextCloud

**Comment by AlexSat** [comment]
I'm really looking forward to this feature in the context of external libraries

**Comment by cooligula** [comment]
Yes, please. This would, I believe, be one of the most useful features altogether. People with Nextcloud, Infomaniak kDrive or similar would have a MUCH better time using this software. It would also enable people without the ability of selfhosting to use this amazing app. Thanks in advance!

**Reply by hannitoball** [reply]
is there a way to connect to ksuite plus over rclone ?

**Comment by CiNcH83** [comment]
Second that. Coming from PhotoPrism and the only thing missing is WebDAV support.

**Comment by justin13888** [comment]
I'm interested in WebDAV not for backup but rather being able to upload my photos, videos, etc. onto Immich before using it directly in my photo and video editing workflows.

Features I'd like is being able to mount a read-only WebDAV for my desktop apps to use or better yet, allow strictly adding new files (e.g. DxO Photolab directly writes your edits into the same folder with option to upload directly).

Personally, based on the current architecture however, I feel like we need to carefully analyze the scope of all the features and assess what is needed in terms of components and technologies to implement this. (Please no subpar webdav implementations.)

**Comment by Atomique** [comment]
I am also interested into an upload option via WebDAV. That would enable me to use PhotoSync on iOS to transfer my stuff to the server.

**Reply by CiNcH83** [reply]
You can also create a SAMBA share and upload to that. PhotoSync supports that. This is how I do it after having switched from PhotoPrism. You can even create multiple shares for multiple users.

**Comment by PersistentCloud** [comment]
I have created a python app for it. Now I can use my iOS tablet and the Pixette app again.

https://github.com/PersistentCloud/immich-webdav-wrapper

Contributions are welcome.

**Comment by m00nwtchr** [comment]
+1, I'd love this in order to be able to use `rclone` to mount Immich for desktop access without opening the web ui.

**Comment by Demian98** [comment]
Just to clarify—Is this request about **WebDAV support for external storages** (e.g., Immich accessing data from something like Nextcloud)?

What I would personally love to see is **WebDAV support where Immich acts as the backend**. This would allow users to sync photos using the best WebDAV-compatible client for their operating system.

For example, I'm currently using Nextcloud to sync my Android gallery (30k+ photos) via the _FolderSync Pro_ app, which works very reliably. Unfortunately, the built-in Immich sync hasn’t been reliable for me and is missing features like **two-way sync**.

It would be fantastic to use Immich with any WebDAV-compatible sync app, making the sync process more flexible and platform-independent.

I’m still new to Immich, so I’d love to hear your thoughts on this idea!

**Comment by FasteryXD** [comment]
WebDAV for example for Nextcloud would be very heavy👍

**Comment by argoyal** [comment]
Any decision taken by the owners to configure external library which is a webdav server?

**Comment by MacroMars** [comment]
Having WebDav as a server (access with compatible clients) would be very good. Having WebDav as a client (Nextcloud as an external library) would be even better. I agree that this is the only thing that PhotoPrism still has over Immich.

**Reply by rstanuwijaya** [reply]
It can save a lot of time for the users migrating from Nextcloud if this can be implemented

**Comment by Demian98** [comment]
I developed a solution (https://github.com/immich-app/immich/discussions/20522) to get this working. Would love to get your feedback to this

**Reply by morbloe** [reply]
This seems to be very nice (didn't try) but it ain't a solution to this feature request. I see one major disadvantage in your approach. If I understand your setup correctly you will end up with the same file two time on your server. (1x in immich, 1x on your sftp-target).

**Reply by Demian98** [reply]
No data is actually stored on the SFTP container. Every access goes live to the Immich API in the background.

Thank you for pointing out on this, I have updated this information as well into the original post.

**Comment by MaluNoPeleke** [comment]
Any updates from the developers on this feature request?

**Reply by bo0tzz** [reply]
https://justinmayer.com/posts/any-updates/

**Comment by fregie** [comment]
This 4-year thread really shows how much people want this.

I ended up going a different route — instead of waiting for Immich to add WebDAV, I built an open source app that connects directly to SMB/WebDAV/NFS storage from your phone. No server-side processing, no hashing, files land on your NAS as-is organized by date. You can browse them from any other tool immediately.

https://github.com/fregie/pho — GPL-3.0, 1.1k stars, iOS + Android. Android APK is free on GitHub Releases.

Not saying it replaces Immich for everyone — it has no AI/ML features, face recognition, etc. But if your main need is "get photos from phone to NAS over WebDAV without anything in between", it does exactly that.

**Reply by Demian98** [reply]
Here is the solution that fixed the problem for me, a SFTP proxy for file sync: https://github.com/Demian98/immich-sftp-server

**Comment by morbloe** [comment]
I find this FR quite vaque! A more concrete one would be: **support webdav as external librarie. 
**
https://github.com/Demian98/immich-sftp-server/releases/tag/v2.0.0
Already provides a very good option IF YOU WANT YOUR PHOTOS TO STAY ON IMMICH. 

But If you want to store your photos on nextcloud, you still have no good option. 
If you go through the hassle of mounting your webdav localy on your immich host, you can provide it somehow. 

But then you are still confronted with the lack of proper folder support in immich for external libraries. And all the other limitations that come with external libraries in immich at the current point.

**Reply by Croydon** [reply]
Yes, being able to keep Photos stored on Nextcloud, but linking them into Immich would be great.
